Code/src/main/java/nl/utwente/viskell/ui/components/LambdaContainer.java

```diff
@@ -145,18 +76,18 @@ public LambdaContainer(DefinitionBlock wrapper, String name, Type type) {
         int i = 0;
         while (t instanceof FunType) {
             FunType ft = (FunType) t;
-            args.add(this.new BinderAnchor(wrapper, new Binder(name + "_x" + i, ft.getArgument())));
+            args.add(new BinderAnchor(this, wrapper, new Binder(name + "_x" + i, ft.getArgument())));
             i++;
             t = ft.getResult();
         }
-        this.res = this.new ResultAnchor(wrapper, Optional.of(t));
+        this.res = new ResultAnchor(this, wrapper, Optional.of(t));
 
         this.argSpace.getChildren().addAll(this.args);
         this.resSpace.getChildren().add(this.res);
     }
 
-    /** Set fresh types in all anchors of this lambda for the next typechecking cycle. */
-    protected void refreshAnchorTypes() {
+    @Override
+    public void refreshAnchorTypes() {
```

Comment:
Unfortunately, you can't have protected methods (or attributes) in an interface. And making it an abstract class is not an option either.